website/docs/language/tests/index.mdx

```diff
@@ -0,0 +1,685 @@
+
+In each `run` block the `expect_failures` attribute can provide a list of checkable objects (resources, data sources, check blocks, input variables, and outputs) that should fail their custom conditions. The test will then pass overall if these checkable objects report an issue, while the test will fail overall if they do not report an issue.
+
+You can still write assertions alongside an `expect_failures` block, but you should be mindful that all custom conditions, except check block assertions, halt the execution of Terraform. This still applies during test execution, so your assertions should only consider values that you are sure will be computed before the checkable object is due to fail. This can be managed via references or the `depends_on` meta-argument within your main configuration.
+
+This also means that, with the exception of `check` blocks, only a single checkable object can be reliably included. We support a list of checkable objects within the `expect_failures` attribute purely for `check` blocks.
+
+A quick example here demonstrates testing the `validation` block on an input variable.
+
+```hcl
+# main.tf
+
+variable "input" {
+  type = number
+
+  validation {
+    condition = var.input % 2 == 0
+    error_message = "must by even number"
+  }
+}
+```
+
+```hcl
+# input_validation.tftest.hcl
+
+variables {
+  input = 0
+}
+
+run "zero" {
+  # The variable defined above is even, so we expect the validation to pass.
+
+  command = plan
+}
+
+run "one" {
+  # This time we set the variable is odd, so we expect the validation to fail.
+
+  command = plan
+
+  variables {
+    input = 1
+  }
+
+  expect_failures = [
+    var.input,
+  ]
+}
+```
+
+>> **Note**: Failures are only expected in the operation specified by the `command` attribute of the `run` block.
+
+This means you should be careful when using `expect_failures` in `run` blocks with `command = apply`. A `run` block with `command = apply` that expects a custom condition failure will fail overall if that custom condition fails during the plan.
+
+This is logically consistent, as the `run` block is expecting to be able to run an apply operation but cannot as the plan failed. It is also potentially confusing, as you will see the failure reported in the diagnostics as the reason the test failed even though that failure was marked as being expected.
+
+There are instances when a custom condition will not be executed during the planning stage as it is relying on computed attributes only available after a referenced resource has been created. In these cases an `expect_failures` block alongside a `command = apply` attribute and value would be expected and acceptable. However, in most cases you should ensure that `expect_failures` are used only alongside `command = plan` operations.
```

Comment:
```suggestion
There are instances when Terraform does not execute a custom condition during the planning stage, because that condition is relying on computed attributes that are only available after Terraform creates the referenced resource. In these cases, you could use an `expect_failures` block alongside a `command = apply` attribute and value. However, in most cases we recommend only using `expect_failures` alongside `command = plan` operations.
```